src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1166,6 +1166,53 @@ internal virtual void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleB
         /// </summary>
         public abstract bool AreLocalsZeroed { get; }
 
+        // PROTOTYPE: Cache the result.
+        internal virtual bool? IsNullableEnabled()
+        {
+            bool? result = false;
+            if (this is SourceMemberMethodSymbol sourceMethod)
+            {
+                result = isNullableEnabled(sourceMethod.SyntaxNode);
+                if (result == true)
+                {
+                    return true;
+                }
+            }
+
+            if (this.IncludeFieldInitializersInBody() &&
+                ContainingType is SourceMemberContainerTypeSymbol containingType)
+            {
+                foreach (var member in containingType.GetMembersUnordered())
+                {
+                    if (member.IsStatic != this.IsStatic)
+                    {
+                        continue;
+                    }
+                    var syntax = member switch
+                    {
+                        SourceFieldSymbolWithSyntaxReference field => field.SyntaxNode,
+                        SourcePropertySymbolBase { BackingField: { } } property => property.SyntaxReference.GetSyntax(),
```

Comment:
>GetSyntax() [](start = 108, length = 11)

Blindly asking for GetSyntax() and then using it feels fragile. We shouldn't assume that whatever is returned is the right thing for the purpose of our check. For example, what do we get for record's positional property here, etc. I really don't want to be guessing here and hoping that we didn't miss some scenario now and worry how new symbols will fit here in the future. I would prefer implementation that makes it very clear what syntax we are looking at. Also, look at ```_lazyMembersAndInitializers``` it looks like we have all interesting syntax nodes readily available.

Reply:
> Also, look at `_lazyMembersAndInitializers` it looks like we have all interesting syntax nodes readily available.

I believe that collection only includes fields that have initializers. `NullableWalker` considers the nullable context of all fields, even those without initializers, to report `ErrorCode.WRN_UninitializedNonNullableField` warnings.

---
In reply to: [545177315](https://github.com/dotnet/roslyn/pull/49953#discussion_r545177315) [](ancestors = 545177315)